source/main.cpp

```diff
@@ -184,7 +184,7 @@ class SocketDemo {
         /* loop until there is nothing received or we've ran out of buffer space */
         nsapi_size_or_error_t result = remaining_bytes;
         while (result > 0 && remaining_bytes > 0) {
```

Comment:
not sure why we even bother checking result, there's an if in the loop that checks for errors

Reply:
I guess we want to terminate the loop on error.

Reply:
There's an if with an early return for that. I guess the difference is if  we got a zero sized packet but that would still be wrong if remaining bytes were >0